web/html/src/manager/systems/bootstrap/bootstrap-minions.js

```diff
@@ -48,12 +51,41 @@ class BootstrapMinions extends React.Component {
         });
     }
 
+    authMethodChanged(event) {
+        this.setState({
+            authMethod: event.target.value
+        });
+    }
+
     passwordChanged(event) {
         this.setState({
             password: event.target.value
         });
     }
 
+    privKeyFileChanged(event) {
+        const file = event.target.files[0];
+        const reader = new FileReader();
+        reader.onload = (e) => this.privKeyLoaded(e.target.result);
+        reader.readAsText(file);
+        this.setState({
+            privKeyLoading: true
```

Comment:
Whenever it turns to use a `flag` as a `loading` traffic-light or condition, I prefer to do it at the very beginning of the function to make sure the info is properly communicated to the user, and to prevent any unexpected behavior from the function body without having the flag set yet. Please move this `setState` right after the `privateKeyFileChanged(event) {` line.